### DISCUSSION TITLE
Symmetry Axis in Axisymmetric Simulation

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#22152] [Q&A Modules: Solid mechanics] [answered]
Hello,

I'm doing an axisymmatric simulation of a ring and want to define the y-axis as the symmetry axis. What I did in former simulations was just using a DirichletBC to stop displacements in x-direction on the left boundary. But now that I have a ring whose left boundary doesn't align with the symmetry axis I can't do that. Is there a way to tell MOOSE that my symmetry axis is 5 mm to the left of my left boundary?

![grafik](https://user-images.githubusercontent.com/98457971/191481048-6754614c-d46b-46d1-9d3b-81b9ecbed668.png)

**Comment by hugary1995** [comment, marked as answer]
I am a little bit confused, because it sounds like you want to utilize the axisymmetry and model a 3D cylinder in 2D, but you are still working with cartesian coordinates, e.g. x, y. Can I assume that you really meant to say the r-displacement instead of the x-displacement?

If so, then the solution will be very simple: you don't prescribe the left boundary in the r-direction.

> Is there a way to tell MOOSE that my symmetry axis is 5 mm to the left of my left boundary?

The best way to achieve that is to have your symmetry axis go through the origin (0, 0). For example, let the bottom left corner of the mesh have an x-coordinate of 5 mm.

**Reply by Flolaffel** [reply]
Yes, you are right. The correct way would be to use r and z. But when using the `TensorMechanicsMasterAction` the Syntax for the variables stays as x and y AFAIK.

Ok so if I understand you correctly MOOSE automatically assumes that there is a symmetry axis through (0,0)? So if I model the geometry as it is pictured above with the bottom left corner as (0,0), I'd fix the r-displacement on the left side of the stemp and the z-displacement on the bottom of the square and that would be it? No assigning a general symmetry axis in z-direction so the program knows the square is supposed to be a ring?

**Reply by hugary1995** [reply]
> Yes, you are right. The correct way would be to use r and z. But when using the `TensorMechanicsMasterAction` the Syntax for the variables stays as x and y AFAIK.

Hmm that's interesting. You can see for example `modules/tensor_mechanics/test/tests/2D_geometries/2D-RZ_test.i` which uses `disp_r`, `disp_z`, `stress_zz` etc. The input file you were looking at probably is probably out-of-date, which we should fix...

> Ok so if I understand you correctly MOOSE automatically assumes that there is a symmetry axis through (0,0)? So if I model the geometry as it is pictured above with the bottom left corner as (0,0), I'd fix the r-displacement on the left side of the stemp and the z-displacement on the bottom of the square and that would be it? No assigning a general symmetry axis in z-direction so the program knows the square is supposed to be a ring?

If the left boundary coincides with the symmetry axis, you may or may not fix the r-displacement there -- the solution will be identical.

If the left boundary does not coincide with the symmetry axis, you should not fix the r-displacement there.

**Reply by Flolaffel** [reply]
> Hmm that's interesting. You can see for example `modules/tensor_mechanics/test/tests/2D_geometries/2D-RZ_test.i` which uses `disp_r`, `disp_z`, `stress_zz` etc. The input file you were looking at probably is probably out-of-date, which we should fix...

Ah well, good to know.

> If the left boundary coincides with the symmetry axis, you may or may not fix the r-displacement there -- the solution will be identical.
> 
> If the left boundary does not coincide with the symmetry axis, you should not fix the r-displacement there.

Ok yeah and the symmetry axis is automatically located on the z-axis?

**Reply by hugary1995** [reply]
Yes, I believe that is a fundamental assumption we make: In tensor_mechanics, the symmetry axis for an axisymmetric cylindrical coordinate system always goes through point (0, 0) and has direction (0, 1), speaking Cartesian.

**Reply by Flolaffel** [reply]
Alright, thank you.
